### ISSUE TITLE
models.metrics

### ISSUE THREAD

**Issue by ahmadmustafaanis** [tensorflow/tensorflow#40030]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 2.2.0
- Python version: 3.7.7
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.1/ V10.1.105
- GPU model and memory: Geforce MX110 2GB

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I created a simple model using tf.keras Sequential API and printed its metrics but it is outputting an empty list. No matter what I try, It returns an empty list.
```
[]
```
**Describe the expected behavior**
It should return 
```
[<tensorflow.python.keras.metrics.BinaryAccuracy at 0x7f3e5e218320>,
 <tensorflow.python.keras.metrics.MeanAbsoluteError at 0x7f3e5c063a20>]
```
**Standalone code to reproduce the issue**
```python3
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Flatten, Conv2D, MaxPooling2D, MaxPool2D
model = Sequential([
    Conv2D(16,(3,3),padding='same', input_shape=(1,28,28),data_format='channels_first'),
    MaxPooling2D((3,3), data_format='channels_first')
])
opt = tf.keras.optimizers.Adam(learning_rate=0.005)
model.compile(optimizer=opt,
              loss=tf.keras.losses.BinaryCrossentropy(),
              metrics=[tf.keras.metrics.BinaryAccuracy(),
              tf.keras.metrics.MeanAbsoluteError()]
              )
print(model.metrics)
```
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by ravikyram** [comment]
I have tried in colab with TF version 2.2, nightly version and was able to reproduce the issue.Please, find the gist [here](https://colab.sandbox.google.com/gist/ravikyram/596178d7832c35f15db66df49febf958/untitled939.ipynb).However i am not seeing any issue with TF version 2.1.0 .Thanks!

**Comment by YinTat** [comment]
Thanks. It is fixed indeed.

**Comment by jvishnuvardhan** [comment]
@ahmadmustafaanis This is expected and was notified in the `Braking Changes` of [2.2.0](https://github.com/tensorflow/tensorflow/releases).

> Keras compile/fit behavior for functional and subclassed models have been unified. Model properties such as metrics, metrics_names will now be available only after training/evaluating the model on actual data for functional models. metrics will now include model loss and output losses.loss_functions property has been removed from the model. This was an undocumented property that was accidentally public and has now been removed.

Please close the issue if this was resolved for you. Thanks!

**Comment by ahmadmustafaanis** [comment]
I tried this in [this](https://colab.research.google.com/gist/ravikyram/596178d7832c35f15db66df49febf958/untitled939.ipynb#scrollTo=kLDcDwDEWAsl) notebook but still, it is not showing metrics.

**Comment by jvishnuvardhan** [comment]
@ahmadmustafaanis As mentioned in the Release notes (Breaking changes), some functionality was unified. It will show the metrics only after training/evaluating the model on **actual data**. Try to run training (`model.fit`) or `model.train_on_batch` and let us know whether you can see the metrics or not. Thanks!

**Comment by ahmadmustafaanis** [comment]
Yes, It is showing on ```model.fit```. Thanks

**Comment by callzhang** [comment]
I still can reproduce the error on tf 2.4.0, using author's code. Any updates on this?

**Comment by jvishnuvardhan** [comment]
@

> @ahmadmustafaanis As mentioned in the Release notes (Breaking changes), some functionality was unified. It will show the metrics only after training/evaluating the model on **actual data**. Try to run training (`model.fit`) or `model.train_on_batch` and let us know whether you can see the metrics or not. Thanks!

@callzhang As mentioned in the above comment, you need to run `model.fit` or `model.train_on_batch` with actual or toy data. Thanks!

**Comment by NEGU93** [comment]
Well, this still looks like an issue to me, I would like to get the metrics without the need to train the model.
